apps/documents/serializers.py

```diff
@@ -0,0 +1,143 @@
+
+    class Meta:
+        model = Document
+        fields = ['name', 'description', 'document_type', 'file_type', 'size', 'upload_status', 'shipment_id']
+
+    def s3_sign(self, document=None):   # pylint: disable=too-many-locals
+        data = self.validated_data
+        s3, _ = get_s3_client()     # pylint: disable=invalid-name
+        bucket = settings.S3_BUCKET
+        file_type = data['file_type']
+        file_type_name = file_type.name.lower()
+
+        # Allowed files type list
+        img_types_list = [enum.name.lower() for enum in FileType]
+        img_types_list.pop(img_types_list.index('pdf'))
+
+        if file_type_name in img_types_list:
+            content_type = f"image/{file_type_name}"
+        else:
+            content_type = f"application/{file_type_name}"
+
+        shipment = document.shipment
+        wallet_id = shipment.shipper_wallet_id
+        storage_credentials_id = shipment.storage_credentials_id
+        vault_id = shipment.vault_id
+        file_s3_path = f"{storage_credentials_id}/{wallet_id}/{vault_id}/{document.id}.{file_type_name}"
+
+        pre_signed_post = s3.generate_presigned_post(
+            Bucket=bucket,
+            Key=file_s3_path,
+            Fields={"acl": "public-read", "Content-Type": content_type},
+            Conditions=[
+                {"acl": "public-read"},
+                {"Content-Type": content_type}
+            ],
+            ExpiresIn=settings.S3_URL_LIFE
+        )
+
+        return {
+            'data': pre_signed_post,
+            'uri': f"s3://{bucket}/{file_s3_path}",
+            'document_id': document.id
+        }
+
+
+class DocumentSerializer(serializers.ModelSerializer):
+    document_type = EnumField(DocumentType, ints_as_names=True)
+    file_type = EnumField(FileType, ints_as_names=True)
+    upload_status = EnumField(UploadStatus, ints_as_names=True)
+
+    class Meta:
+        model = Document
+        exclude = ['shipment']
+        read_only_fields = ('owner_id', 'shipment', 'document_type', 'file_type', 'size', 's3_path')
+
+    class JSONAPIMeta:
+        included_resources = ['shipment']
+
+
+class DocumentUpdateSerializer(serializers.ModelSerializer):
+    upload_status = EnumField(UploadStatus, ints_as_names=True)
+
+    class Meta:
+        model = Document
+        fields = '__all__'
+        read_only_fields = ('owner_id', 'shipment', 'document_type', 'file_type', 'size', 's3_path')
+
+
+class DocumentListSerializer(serializers.ListSerializer):
+    """
+    This custom class allows us to add to the returned documents the one time s3 generated url
+    """
+    @property
+    def data(self):
+        # pass
+        data = super(DocumentListSerializer, self).data
+        super(DocumentListSerializer, self).data        # pylint: disable=expression-not-assigned
+
+        for item in data:
+            item.update({'url': Document.objects.get(id=item['id']).generate_presigned_url})
```

Comment:
Instead of doing this, I'd recommend making this a ModelSerializer and including the presigned_url field.